drivers/spi/spi_nrfx_spim.c

```diff
@@ -346,13 +339,14 @@ static int spim_nrfx_pm_control(const struct device *dev,
 
 	switch (action) {
 	case PM_DEVICE_ACTION_RESUME:
-		ret = init_spim(dev);
-		/* Force reconfiguration before next transfer */
-		data->ctx.config = NULL;
+		/* No action needed at this point, nrfx_spim_init() will be
+		 * called at configuration before the next transfer.
+		 */
```

Comment:
PM hooks are no longer doing what is expected, ie, put the device into operational state when resume is called. Why is this change required?

Reply:
Previously, when resume was called, the device was brought to the state it was right after its initialization - it still needed to be reconfigured before the next transfer:
https://github.com/zephyrproject-rtos/zephyr/blob/7ccc1a41bc4c27c73bb03f01ed6b05491ba5872d/drivers/spi/spi_nrfx_spim.c#L336-L338
(at the device initialization, the nrfx driver was initialized with a default configuration, which was then altered accordingly when a transfer was requested). Now the difference is that the initialization of the device does not include the initialization of the nrfx driver - this is done when the device is configured right before a transfer is performed (only then we know the frequency, SPI mode, etc. that should be used), so there is no point in calling `nrfx_spim_init()` here and then, right before the next transfer, calling `nrfx_spim_uninit()` in order to be able to call `nrfx_spim_init()` again.
Alternatively, I would need to store the currently used nrfx driver configuration to be able to use it here in the call to `nrfx_spim_init()`. Then it would be not needed to reconfigure the device before the next transfer, provided it would be requested with the same configuration. But I'm not sure if it is worth spending more RAM just to achieve this possibility.